airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGeneratorIntegrationTest.java

```diff
@@ -851,6 +856,48 @@ public void softReset() throws InterruptedException {
     assertEquals(0, rawUntypedRows);
   }
 
+  @Test
+  public void testMigrateFromV1toV2() throws InterruptedException {
+    createV1RawTable();
+    final StreamId aStreamId = new StreamId(null, null, testDataset, "_users_v2_raw", null, null);
+    final StreamConfig stream = new StreamConfig(aStreamId, null, null, null, null, null);
+    final AirbyteStreamNameNamespacePair v1RawNameNamespacePair = new AirbyteStreamNameNamespacePair("_users_v1_raw", testDataset);
+    // Load data into v1 raw table
+    Map<String, String> replacementValues = new HashMap<>();
+    replacementValues.put("dataset", testDataset);
+    IntStream.range(0, 2).forEach(i -> replacementValues.put("id_%d".formatted(i), UUID.randomUUID().toString()));
+    bqExecuteQuery(replacementValues, """
+        INSERT INTO ${dataset}._users_v1_raw (`_airbyte_ab_id`, `_airbyte_data`, `_airbyte_emitted_at`) VALUES
+        ('${id_0}', '{"hello": "world"}', '2023-01-01T00:00:00Z'),
+        ('${id_1}', '{"hello": "world"}', '2023-01-01T00:00:00Z');
+        """);
+    final String migration = GENERATOR.migrateFromV1toV2(stream, v1RawNameNamespacePair);
+    destinationHandler.execute(migration);
+    // Get v1 raw table, make sure it still exists
+    final var v1RawRows = bqExecuteQuery(Map.of("dataset", testDataset), "SELECT * FROM ${dataset}._users_v1_raw;").getTotalRows();
+    assertEquals(2, v1RawRows);
+    // Get v2 raw table, make sure it exists and has the correct schema
+    final var v2RawRows = bqExecuteQuery(Map.of("dataset", testDataset), "SELECT * FROM ${dataset}._users_v2_raw;").getTotalRows();
+    assertEquals(2, v2RawRows);
```

Comment:
this assert is probably easier using `BaseSqlGeneratorIntegrationTest#verifyRecords` (plus it'll allow verifying the exact record contents)